lib/ansible/modules/web_infrastructure/ansible_tower/tower_project.py

```diff
@@ -202,7 +207,9 @@ def main():
                 json_output['id'] = result['id']
             elif state == 'absent':
                 result = project.delete(name=name)
-        except (exc.ConnectionError, exc.BadRequest) as excinfo:
+            elif state == 'updated':
+                result = project.update(name=name, wait=True)
```

Comment:
What if `state: updated` is given, but the resource doesn't yet exist?

Shouldn't it be created?

Reply:
I don't know that we want "updated" to imply "present".

The behavior right now is we just return NotFound in this case.